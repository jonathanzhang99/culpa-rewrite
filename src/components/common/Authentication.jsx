import PropTypes from "prop-types";
import React, { useContext, useReducer } from "react";
import { Route, Redirect } from "react-router-dom";

export const AuthContext = React.createContext(null);

function authReducer(state, action) {
  switch (action.type) {
    case "LOGIN_START":
      return {
        ...state,
        isLoading: true,
      };
    case "LOGIN_ERROR":
      return {
        ...state,
        isError: true,
        isLoading: false,
      };

    case "LOGIN_SUCCESS":
      localStorage.setItem("authToken", JSON.stringify(action.payload));
      return {
        ...state,
        isError: false,
        isLoading: false,
        isLoggedIn: true,
        authToken: action.payload,
      };

    case "LOGIN_FAILURE":
      return {
        ...state,
        isError: false,
        isLoading: false,
        isLoggedIn: false,
      };
    case "LOGOUT_SUCCESS":
      localStorage.removeItem("authToken");
      return {
        ...state,
        isError: false,
        isLoading: false,
        isLoggedIn: false,
        authToken: undefined,
      };
    default:
      throw new Error();
  }
}

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

/**
 * AuthProvider createa React Context with values such that children elements
 * in the tree can access user auth token and manipulate them
 * (e.g. login, logout)
 *
 */
export function AuthProvider({ children }) {
  const existingToken = JSON.parse(localStorage.getItem("authToken"));

  const [state, dispatch] = useReducer(authReducer, {
    isError: false,
    isLoading: false,
    isLoggedIn: !!existingToken,
    authToken: existingToken,
  });

  // `login` either returns an error or null if a token was succesfully
  // retrieved. The error will always be a string message
  const login = async ({ username, password }) => {
    dispatch({ type: "LOGIN_START" });
    const response = await fetch("/api/auth/login", {
      method: "POST",
      body: JSON.stringify({
        username,
        password,
      }),
      headers: {
        "Content-Type": "application/json",
      },
    });
    try {
      const result = await response.json();

      // Require that the backend sends either a token or an error
      result.error
        ? dispatch({ type: "LOGIN_FAILURE" })
        : dispatch({ type: "LOGIN_SUCCESS", payload: result.token });
      return result;
    } catch (err) {
      dispatch({ type: "LOGIN_ERROR" });
      return err;
    }
  };

  // `logout` simply removes the token from the client. It does NOT
  // expire the token.
  const logout = async () => {
    if (state.authToken) {
      dispatch({ type: "LOGOUT_SUCCESS" });
    }
  };

  return (
    <AuthContext.Provider
      value={{
        ...state,
        login,
        logout,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
}

AuthProvider.propTypes = propTypes;

/**
 * ProtectedRoute allows access to the url only if a user is logged in.
 *
 */
export function ProtectedRoute(props) {
  const { isLoggedIn } = useContext(AuthContext);
  // eslint-disable-next-line react/jsx-props-no-spreading
  return isLoggedIn ? <Route {...props} /> : <Redirect to="/" />;
}
