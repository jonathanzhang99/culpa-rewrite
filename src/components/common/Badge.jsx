import PropTypes from "prop-types";
import React from "react";
import { Image } from "semantic-ui-react";

import bronzeNugget from "icons/bronze-nugget.svg";
import goldNugget from "icons/gold-nugget.svg";
import silverNugget from "icons/silver-nugget.svg";

const propTypesBadge = {
  badgeId: PropTypes.number.isRequired,
};

export default function Badge({ badgeId }) {
  switch (badgeId) {
    case 1: {
      return <Image avatar spaced="left" src={goldNugget} />;
    }
    case 2: {
      return <Image avatar spaced="left" src={silverNugget} />;
    }
    case 3: {
      return <Image avatar spaced="left" src={bronzeNugget} />;
    }
    default: {
      return null;
    }
  }
}

Badge.propTypes = propTypesBadge;
