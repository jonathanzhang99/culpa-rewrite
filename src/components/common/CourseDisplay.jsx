import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { Header } from "semantic-ui-react";

const propTypesCourseDisplayName = {
  as: PropTypes.oneOf(["header", "text"]),
  courseCallNumber: PropTypes.string,
  courseName: PropTypes.string.isRequired,
  size: PropTypes.oneOf(["tiny", "small", "medium", "large", "huge"]),
};

const propTypesCourseDisplayLink = {
  as: PropTypes.oneOf(["header", "text"]),
  courseCallNumber: PropTypes.string,
  courseId: PropTypes.number.isRequired,
  courseName: PropTypes.string.isRequired,
  size: PropTypes.oneOf(["tiny", "small", "medium", "large", "huge"]),
};

const defaultProps = {
  courseCallNumber: "",
  as: "text",
  size: "medium",
};

export function CourseDisplayName({ as, courseCallNumber, courseName, size }) {
  const CourseName = as === "header" ? Header : "span";
  return (
    <CourseName size={size}>
      {courseCallNumber} {courseName}
    </CourseName>
  );
}

export function CourseDisplayLink({
  as,
  courseCallNumber,
  courseId,
  courseName,
  size,
}) {
  return (
    <Link to={`/course/${courseId}`}>
      <CourseDisplayName
        as={as}
        courseCallNumber={courseCallNumber}
        courseName={courseName}
        size={size}
      />
    </Link>
  );
}

CourseDisplayName.propTypes = propTypesCourseDisplayName;
CourseDisplayName.defaultProps = defaultProps;

CourseDisplayLink.propTypes = propTypesCourseDisplayLink;
CourseDisplayLink.defaultProps = defaultProps;
