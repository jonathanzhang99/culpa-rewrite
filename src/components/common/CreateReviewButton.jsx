/* eslint-disable react/jsx-props-no-spreading */
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { Button } from "semantic-ui-react";

const propTypes = {
  color: PropTypes.string,
  courseId: PropTypes.number,
  fluid: PropTypes.bool,
  subject: PropTypes.string,
  professorId: PropTypes.number,
  relaxed: PropTypes.bool,
};

const defaultProps = {
  color: "orange",
  courseId: undefined,
  fluid: false,
  subject: "",
  professorId: undefined,
  relaxed: false,
};

export default function CreateReviewButton({
  color,
  courseId,
  fluid,
  professorId,
  subject,
  relaxed,
}) {
  return (
    <Link
      to={{
        pathname: "/review/submit",
        state: { courseId, professorId },
      }}
    >
      <Button
        className={`${relaxed ? "relaxed" : ""} create-review-button`}
        color={color}
        fluid={fluid}
      >
        WRITE A REVIEW {subject ? `FOR ${subject.toUpperCase()}` : ""}
      </Button>
    </Link>
  );
}

CreateReviewButton.propTypes = propTypes;
CreateReviewButton.defaultProps = defaultProps;
