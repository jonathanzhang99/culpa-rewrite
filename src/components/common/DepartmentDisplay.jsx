import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { Header } from "semantic-ui-react";

const propTypesDepartmentDisplayName = {
  as: PropTypes.oneOf(["header", "text"]),
  departmentName: PropTypes.string.isRequired,
  size: PropTypes.oneOf(["tiny", "small", "medium", "large", "huge"]),
};

const propTypesDepartmentDisplayLink = {
  as: PropTypes.oneOf(["header", "text"]),
  departmentId: PropTypes.number.isRequired,
  departmentName: PropTypes.string.isRequired,
  size: PropTypes.oneOf(["tiny", "small", "medium", "large", "huge"]),
};

const defaultProps = {
  as: "text",
  size: "medium",
};

export function DepartmentDisplayName({ as, departmentName, size }) {
  const DepartmentName = as === "header" ? Header : "span";
  return <DepartmentName size={size}> {departmentName} </DepartmentName>;
}

export function DepartmentDisplayLink({
  as,
  departmentId,
  departmentName,
  size,
}) {
  return (
    <Link to={`/department/${departmentId}`}>
      <DepartmentDisplayName
        as={as}
        departmentName={departmentName}
        size={size}
      />
    </Link>
  );
}

DepartmentDisplayName.propTypes = propTypesDepartmentDisplayName;
DepartmentDisplayName.defaultProps = defaultProps;

DepartmentDisplayLink.propTypes = propTypesDepartmentDisplayLink;
DepartmentDisplayLink.defaultProps = defaultProps;
