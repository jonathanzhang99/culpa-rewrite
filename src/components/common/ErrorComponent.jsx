import React from "react";

export default function ErrorComponent() {
  return <div>error occurred</div>;
}
