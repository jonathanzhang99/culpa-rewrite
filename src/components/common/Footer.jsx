import React from "react";
import { Link } from "react-router-dom";
import { Grid, Icon, Image } from "semantic-ui-react";

import culpaIcon from "icons/culpa.svg";

export default function Footer() {
  return (
    <Grid className="footer" columns={2} verticalAlign="middle">
      <Grid.Column textAlign="center">
        <Image centered src={culpaIcon} />
        <Icon color="blue" name="copyright outline" size="small" />
        <span style={{ color: "#82B7E8" }}>1997-2020</span>
      </Grid.Column>
      <Grid.Column textAlign="left">
        <Grid.Row>
          <Link className="footer-link" to="/">
            About
          </Link>
        </Grid.Row>
        <Grid.Row>
          <Link className="footer-link" to="/join-the-team">
            Join The Team
          </Link>
        </Grid.Row>
        <Grid.Row>
          <Icon color="blue" name="mail" size="small" />
          <span className="footer-link">admin@culpa.info</span>
        </Grid.Row>
      </Grid.Column>
    </Grid>
  );
}
