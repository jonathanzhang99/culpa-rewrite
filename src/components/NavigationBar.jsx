import PropTypes, { oneOfType } from "prop-types";
import React, { useState } from "react";
import { useHistory, Link } from "react-router-dom";
import { Button, Container, Grid, Icon, Image, Menu } from "semantic-ui-react";

import CreateReviewButton from "components/common/CreateReviewButton";
import Footer from "components/common/Footer";
import Form from "components/common/Form";
import { SearchInput } from "components/common/Inputs";
import Sidebar from "components/common/Sidebar";
import culpaIcon from "icons/culpa.svg";

const propTypes = {
  children: oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
  ]),
};

const defaultProps = {
  children: [],
};

export default function NavigationBar({ children }) {
  const [isSidebarVisible, setSidebarVisible] = useState(false);

  const showSidebar = () => {
    setSidebarVisible(true);
  };

  const hideSidebar = () => {
    setSidebarVisible(false);
  };

  const history = useHistory();

  const onSearchSubmit = ({ NavbarSearchbar }) => {
    const searchValue = NavbarSearchbar;
    if (typeof searchValue === "string" && searchValue.length > 1) {
      history.push(`/search?entity=all&query=${searchValue}&alphabetize=True`);
    }
    return {}; // Form requires output
  };

  const onResultSelect = ({ type, id }) => {
    history.push(`/${type}/${id}`);
  };

  return (
    <>
      <Grid
        as={Menu}
        className="navbar"
        fixed="top"
        textAlign="center"
        verticalAlign="middle"
      >
        <Grid.Column width={1}>
          <Button basic compact onClick={showSidebar}>
            <Icon fitted color="blue" name="bars" size="big" />
          </Button>
        </Grid.Column>
        <Grid.Column width={3}>
          <Link to="/">
            <Image src={culpaIcon} />
          </Link>
        </Grid.Column>
        <Grid.Column width={9}>
          <Form onSubmit={onSearchSubmit} onSuccess={() => {}}>
            <SearchInput
              name="NavbarSearchbar"
              placeholder="Search for professors or courses"
              searchEntity="all"
              searchLimit={6}
              onResultSelect={onResultSelect}
            />
          </Form>
        </Grid.Column>
        <Grid.Column width={3}>
          <CreateReviewButton fluid color="orange" />
        </Grid.Column>
      </Grid>
      <Container className="root-container">
        <Sidebar hideSidebar={hideSidebar} isSidebarVisible={isSidebarVisible}>
          {children}
        </Sidebar>
      </Container>
      <Footer />
    </>
  );
}

NavigationBar.propTypes = propTypes;
NavigationBar.defaultProps = defaultProps;
