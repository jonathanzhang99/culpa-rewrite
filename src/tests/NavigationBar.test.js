import { render, fireEvent, act, screen, wait } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { AuthProvider } from "components/common/Authentication";
import NavigationBar from "components/NavigationBar";

describe("Navbar Component Tests", () => {
  const professorAndCourseResults = {
    professorResults: [
      {
        childKey: "professor-2339",
        departments: [
          {
            id: 29,
            name: "Computer Science",
          },
        ],
        last: "true",
        id: 2339,
        title: "Nakul Verma",
        type: "professor",
      },
    ],
    courseResults: [
      {
        childKey: "course-1",
        departments: [
          {
            id: 6,
            name: "Computer Science",
          },
        ],
        id: 1,
        title: "User Interface Design",
        type: "course",
      },
    ],
  };

  const onlyProfessorResults = {
    professorResults: [
      {
        childKey: "professor-2339",
        departments: [
          {
            id: 29,
            name: "Computer Science",
          },
        ],
        last: "true",
        id: 2339,
        title: "Nakul Verma",
        type: "professor",
      },
    ],
    courseResults: [],
  };

  const onlyCourseResults = {
    professorResults: [],
    courseResults: [
      {
        childKey: "course-1",
        departments: [
          {
            id: 6,
            name: "Computer Science",
          },
        ],
        id: 1,
        title: "User Interface Design",
        type: "course",
      },
    ],
  };

  const noResults = {
    professorResults: [],
    courseResults: [],
  };

  const serverFoundProfessorAndCourseResults = () =>
    Promise.resolve({
      ok: true,
      json: () => professorAndCourseResults,
    });

  const serverFoundOnlyProfessorResults = () =>
    Promise.resolve({
      ok: true,
      json: () => onlyProfessorResults,
    });

  const serverFoundOnlyCourseResults = () =>
    Promise.resolve({
      ok: true,
      json: () => onlyCourseResults,
    });

  const serverFoundNoResults = () =>
    Promise.resolve({
      ok: true,
      json: () => noResults,
    });

  test("should render", () => {
    const snapshot = render(
      <AuthProvider>
        <MemoryRouter>
          <NavigationBar />
        </MemoryRouter>
      </AuthProvider>
    );
    expect(snapshot).toMatchSnapshot();
  });

  describe("searchbar interactions", () => {
    let mockFetch;

    beforeEach(() => {
      mockFetch = jest.spyOn(global, "fetch");
      render(
        <AuthProvider>
          <MemoryRouter>
            <NavigationBar />
          </MemoryRouter>
        </AuthProvider>
      );
    });

    afterEach(() => jest.resetAllMocks());

    test("both professor and course results found", async () => {
      mockFetch.mockImplementation(serverFoundProfessorAndCourseResults);

      await act(async () => {
        fireEvent.input(screen.getByRole("textbox"), {
          target: { value: "testSearchValue" },
        });
      });

      // waits until debouncer fetches search results
      await wait(() => {
        expect(screen.getByText(/nakul verma/i)).toBeInTheDocument();
        expect(screen.getByText(/user interface design/i)).toBeInTheDocument();
        expect(mockFetch).toHaveBeenCalledWith(
          "/api/search?entity=all&query=testSearchValue&limit=7",
          {
            method: "GET",
            headers: { "Content-Type": "Application/json" },
          }
        );
      });
    });

    test("only professor results found", async () => {
      mockFetch.mockImplementation(serverFoundOnlyProfessorResults);

      await act(async () => {
        fireEvent.input(screen.getByRole("textbox"), {
          target: { value: "testSearchValue" },
        });
      });

      await wait(() => {
        expect(screen.getByText(/nakul verma/i)).toBeInTheDocument();
        expect(
          screen.queryByText(/user interface design/i)
        ).not.toBeInTheDocument();
        expect(mockFetch).toHaveBeenCalledWith(
          "/api/search?entity=all&query=testSearchValue&limit=7",
          {
            method: "GET",
            headers: { "Content-Type": "Application/json" },
          }
        );
      });
    });

    test("only course results found", async () => {
      mockFetch.mockImplementation(serverFoundOnlyCourseResults);

      await act(async () => {
        fireEvent.input(screen.getByRole("textbox"), {
          target: { value: "testSearchValue" },
        });
      });

      await wait(() => {
        expect(screen.queryByText(/nakul verma/i)).not.toBeInTheDocument();
        expect(screen.getByText(/user interface design/i)).toBeInTheDocument();
        expect(mockFetch).toHaveBeenCalledWith(
          "/api/search?entity=all&query=testSearchValue&limit=7",
          {
            method: "GET",
            headers: { "Content-Type": "Application/json" },
          }
        );
      });
    });

    test("no results found", async () => {
      mockFetch.mockImplementation(serverFoundNoResults);

      await act(async () => {
        fireEvent.input(screen.getByRole("textbox"), {
          target: { value: "testSearchValue" },
        });
      });

      await wait(() => {
        expect(screen.queryByText(/nakul verma/i)).not.toBeInTheDocument();
        expect(
          screen.queryByText(/user interface design/i)
        ).not.toBeInTheDocument();
        expect(mockFetch).toHaveBeenCalledWith(
          "/api/search?entity=all&query=testSearchValue&limit=7",
          {
            method: "GET",
            headers: { "Content-Type": "Application/json" },
          }
        );
      });
    });
  });
});
