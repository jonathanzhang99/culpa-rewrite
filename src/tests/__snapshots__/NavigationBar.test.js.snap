// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui top fixed ui center aligned middle aligned grid navbar menu"
      >
        <div
          class="one wide column"
        >
          <button
            class="ui basic compact button"
          >
            <i
              aria-hidden="true"
              class="blue bars big fitted icon"
            />
          </button>
        </div>
        <div
          class="three wide column"
        >
          <a
            href="/"
          >
            <img
              class="ui image"
              src="culpa.svg"
            />
          </a>
        </div>
        <div
          class="nine wide column"
        >
          <form
            class="ui form"
          >
            <div
              class="ui error message"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                >
                  Error
                </div>
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="ui fluid search"
              >
                <div
                  class="ui icon input"
                >
                  <input
                    aria-invalid="true"
                    autocomplete="off"
                    class="prompt"
                    id="form-input-NavbarSearchbar"
                    name="NavbarSearchbar"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <i
                    aria-hidden="true"
                    class="search icon"
                  />
                </div>
                <div
                  class="results transition"
                >
                  <div
                    class="message empty"
                  >
                    <div
                      class="header"
                    >
                      No results found
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          class="three wide column"
        >
          <a
            href="/review/submit"
          >
            <button
              class="ui orange fluid button  create-review-button"
            >
              WRITE A REVIEW 
              
            </button>
          </a>
        </div>
      </div>
      <div
        class="ui container root-container"
      >
        <div
          class="ui borderless vertical ui overlay left sidebar menu"
        >
          <a
            class="item sidebar-header"
            href="/"
          >
            WELCOME TO CULPA!
          </a>
          <a
            class="item"
            href="/"
          >
            Home
            <i
              aria-hidden="true"
              class="blue angle right large icon"
            />
          </a>
          <a
            class="item"
            href="/about"
          >
            About
            <i
              aria-hidden="true"
              class="blue angle right large icon"
            />
          </a>
          <a
            class="item"
            href="/departments"
          >
            Departments
            <i
              aria-hidden="true"
              class="blue angle right large icon"
            />
          </a>
          <a
            class="item"
            href="/join-the-team"
          >
            Join the Team
            <i
              aria-hidden="true"
              class="blue angle right large icon"
            />
          </a>
          <a
            class="item"
            href="/login"
          >
            Login
            <i
              aria-hidden="true"
              class="blue angle right large icon"
            />
          </a>
        </div>
        <div
          class="pushable"
        >
          <div
            class="pusher"
          />
        </div>
      </div>
      <div
        class="ui middle aligned two column grid footer"
      >
        <div
          class="center aligned column"
        >
          <img
            class="ui centered image"
            src="culpa.svg"
          />
          <i
            aria-hidden="true"
            class="blue copyright outline small icon"
          />
          <span
            style="color: rgb(130, 183, 232);"
          >
            1997-2020
          </span>
        </div>
        <div
          class="left aligned column"
        >
          <div
            class="row"
          >
            <a
              class="footer-link"
              href="/"
            >
              About
            </a>
          </div>
          <div
            class="row"
          >
            <a
              class="footer-link"
              href="/join-the-team"
            >
              Join The Team
            </a>
          </div>
          <div
            class="row"
          >
            <i
              aria-hidden="true"
              class="blue mail small icon"
            />
            <span
              class="footer-link"
            >
              admin@culpa.info
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui top fixed ui center aligned middle aligned grid navbar menu"
    >
      <div
        class="one wide column"
      >
        <button
          class="ui basic compact button"
        >
          <i
            aria-hidden="true"
            class="blue bars big fitted icon"
          />
        </button>
      </div>
      <div
        class="three wide column"
      >
        <a
          href="/"
        >
          <img
            class="ui image"
            src="culpa.svg"
          />
        </a>
      </div>
      <div
        class="nine wide column"
      >
        <form
          class="ui form"
        >
          <div
            class="ui error message"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                Error
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui fluid search"
            >
              <div
                class="ui icon input"
              >
                <input
                  aria-invalid="true"
                  autocomplete="off"
                  class="prompt"
                  id="form-input-NavbarSearchbar"
                  name="NavbarSearchbar"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <i
                  aria-hidden="true"
                  class="search icon"
                />
              </div>
              <div
                class="results transition"
              >
                <div
                  class="message empty"
                >
                  <div
                    class="header"
                  >
                    No results found
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="three wide column"
      >
        <a
          href="/review/submit"
        >
          <button
            class="ui orange fluid button  create-review-button"
          >
            WRITE A REVIEW 
            
          </button>
        </a>
      </div>
    </div>
    <div
      class="ui container root-container"
    >
      <div
        class="ui borderless vertical ui overlay left sidebar menu"
      >
        <a
          class="item sidebar-header"
          href="/"
        >
          WELCOME TO CULPA!
        </a>
        <a
          class="item"
          href="/"
        >
          Home
          <i
            aria-hidden="true"
            class="blue angle right large icon"
          />
        </a>
        <a
          class="item"
          href="/about"
        >
          About
          <i
            aria-hidden="true"
            class="blue angle right large icon"
          />
        </a>
        <a
          class="item"
          href="/departments"
        >
          Departments
          <i
            aria-hidden="true"
            class="blue angle right large icon"
          />
        </a>
        <a
          class="item"
          href="/join-the-team"
        >
          Join the Team
          <i
            aria-hidden="true"
            class="blue angle right large icon"
          />
        </a>
        <a
          class="item"
          href="/login"
        >
          Login
          <i
            aria-hidden="true"
            class="blue angle right large icon"
          />
        </a>
      </div>
      <div
        class="pushable"
      >
        <div
          class="pusher"
        />
      </div>
    </div>
    <div
      class="ui middle aligned two column grid footer"
    >
      <div
        class="center aligned column"
      >
        <img
          class="ui centered image"
          src="culpa.svg"
        />
        <i
          aria-hidden="true"
          class="blue copyright outline small icon"
        />
        <span
          style="color: rgb(130, 183, 232);"
        >
          1997-2020
        </span>
      </div>
      <div
        class="left aligned column"
      >
        <div
          class="row"
        >
          <a
            class="footer-link"
            href="/"
          >
            About
          </a>
        </div>
        <div
          class="row"
        >
          <a
            class="footer-link"
            href="/join-the-team"
          >
            Join The Team
          </a>
        </div>
        <div
          class="row"
        >
          <i
            aria-hidden="true"
            class="blue mail small icon"
          />
          <span
            class="footer-link"
          >
            admin@culpa.info
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
